### PR TITLE
Events support

### PR DESCRIPTION
**Pull request by kalaninja**

### CODE REVIEW

File: crypto/src/main/java/com/strategyobject/substrateclient/crypto/Hasher.java [hunk 1]
```diff
@@ -16,7 +16,8 @@ public static byte[] blake2(Size size, byte[] value) {
     }
 
     public static long xx(long seed, byte[] value) {
-        return LongHashFunction.xx(seed).hashBytes(value);
+        val f = LongHashFunction.xx(seed);
```
Comment (vnabiev) [inline crypto/src/main/java/com/strategyobject/substrateclient/crypto/Hasher.java:19]: What is a purpose of that?

File: api/src/main/java/com/strategyobject/substrateclient/api/DefaultModule.java [hunk 2]
```diff
@@ -97,8 +109,13 @@ protected void configure() {
 
     @Provides
     @Singleton
-    public ScaleReaderRegistry provideScaleReaderRegistry() {
+    public ScaleReaderRegistry provideScaleReaderRegistry(MetadataProvider metadataProvider,
+                                                          EventRegistry eventRegistry) {
         val registry = new ScaleReaderRegistry();
+        registry.registerAnnotatedFrom(PREFIX);
+        registry.register(new AccountDataReader(registry), System.AccountData.class, AccountData.class);
```
Comment (vnabiev) [inline api/src/main/java/com/strategyobject/substrateclient/api/DefaultModule.java:116]: Could it be registered automatically?

File: crypto/src/main/java/com/strategyobject/substrateclient/crypto/Hasher.java (hunk 1 above)
Comment (kalaninja) [reply crypto/src/main/java/com/strategyobject/substrateclient/crypto/Hasher.java:19]: it is an artifact. I noticed an issue in LongHashFunction and took a look into it.

File: api/src/main/java/com/strategyobject/substrateclient/api/DefaultModule.java (hunk 2 above)
Comment (kalaninja) [reply api/src/main/java/com/strategyobject/substrateclient/api/DefaultModule.java:116]: no, unless write a custom scale reader.

**Review by vnabiev — APPROVED** [review]

**Pull request merged**